src/torchmetrics/functional/regression/pearson.py

```diff
@@ -43,18 +44,24 @@ def _pearson_corrcoef_update(
     """
     # Data checking
     _check_same_shape(preds, target)
-    preds = preds.squeeze()
-    target = target.squeeze()
-    if preds.ndim > 1 or target.ndim > 1:
-        raise ValueError("Expected both predictions and target to be 1 dimensional tensors.")
-
-    n_obs = preds.numel()
-    mx_new = (n_prior * mean_x + preds.mean() * n_obs) / (n_prior + n_obs)
-    my_new = (n_prior * mean_y + target.mean() * n_obs) / (n_prior + n_obs)
+    if preds.ndim > 2 or target.ndim > 2:
+        raise ValueError(
+            f"Expected both predictions and target to be either 1- or 2-dimensional tensors,"
+            f" but got {target.ndim} and {preds.ndim}."
+        )
+    if (n_out == 1 and preds.ndim != 1) or (n_out > 1 and n_out != preds.shape[-1]):
+        raise ValueError(
+            f"Expected argument `num_outputs` to match the second dimension of input, but got {n_out}"
```

Comment:
Naming inconsistency - the argument is called `n_out` and here we refer to `num_outputs`. Personally, I like the latter maybe a little bit more, but I'm relatively indifferent here.

Also, `n_out` is not a part of public API hence I'm a bit afraid of it might be confusing for users to receive such an error message.